### PR TITLE
box: export current session's identifier to C API

### LINKED ISSUE #7895: Export current session ID to C API
In order to send IPROTO packets using `box_iproto_send` (gh-7897) we need to have some session identifier source in the C API: for now, export an analog of `box.session.id`.

# Interface

Returns the current session's identifier (see also [`box.session.id`](https://github.com/tarantool/tarantool/blob/dec0e0221e183fa972efa65bb0fb658112f2196f/src/box/lua/session.c#L75-L88)).
```C
uint64_t
box_session_id();
```

### PR DESCRIPTION
**Pull request by CuriousGeorgiy**
Closes #7895

### CODE REVIEW

File: test/app-tap/module_api.c [hunk 1]
```diff
@@ -3045,6 +3058,7 @@ luaopen_module_api(lua_State *L)
 		{"isdecimal", test_isdecimal},
 		{"isdecimal_ptr", test_isdecimal_ptr},
 		{"box_schema_version_matches", test_box_schema_version},
+		{"box_session_id", test_box_session_id},
```
Comment (locker) [inline test/app-tap/module_api.c:3061]: Nit: Please rename "box_session_id" to "box_session_id_matches" here for consistency with "box_schema_version_matches".

File: src/box/box.h [hunk 2]
```diff
@@ -599,6 +599,12 @@ box_sequence_reset(uint32_t seq_id);
 API_EXPORT int
 box_session_push(const char *data, const char *data_end);
 
+/**
+ * \return current session's unique monotonic identifier (\sa box.session.id)
+ */
+API_EXPORT uint64_t
+box_session_id();
```
Comment (locker) [inline src/box/box.h:606]: Nit: Should be defined/declared as `box_session_id(void)` (otherwise it can take an arbitrary number of arguments in C).

Comment (CuriousGeorgiy) [reply src/box/box.h]: Fixed: added `void` type parameter.

File: test/app-tap/module_api.c (hunk 1 above)
Comment (CuriousGeorgiy) [reply test/app-tap/module_api.c]: Fixed: rename all test related names similar to `box_schema_version`.

**Review by locker — APPROVED** [review]

**Pull request merged**